migrate various modules from UserPreferences to Proto-DataStore

rhkean opened the pull request:
* migrate below modules from SharedPreferences and Preferences-Datastore to a common Proto-DataStore Settings object in the core module
    - permissions-ble,
    - permissions-notification,
    - permissions-wifi
    - analytics
* aligned `permissions-*` to have the same file/package structure
* updated all libraries to the latest (as of 8/1/2025)

rhkean committed:
migrate
* permissions-ble,
* permissions-notification,
* permissions-wifi
from UserPreferences to Proto-DataStore Preferences

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/NordicPlayground/Android-Common-Libraries?pullRequest=136) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/NordicPlayground/Android-Common-Libraries?pullRequest=136) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/NordicPlayground/Android-Common-Libraries?pullRequest=136) it.</sub>

rhkean committed:
bad cut and paste, updating wrong setting property for notifications

rhkean commented:
I've got something wrong with a scope or something.  every thing appears to be working, except the actual writes to the datastore file.  :/

I put the PR in draft until I can resolve it

rhkean commented:
I've never had this problem before... might be a multiple module scenario or something....  maybe a scope closing before it's completed.... very weird.
The datastore file is created, but the write-to method on the Serializer never executes

rhkean committed:
working, but still not saving value

rhkean committed:
datastore correctly working

implemented singleton repository that does not require Hilt

rhkean committed:
import cleanup
rearranged package layout to standardize them for permissions-ble, permissions-notification, & permissions-wifi

rhkean commented:
fixed!

rhkean committed:
spelling error in filename

rhkean committed:
update all libraries

rhkean committed:
Merge remote-tracking branch 'origin/migrate_to_proto-datastore' into migrate_to_proto-datastore

rhkean committed:
correcting library versions

rhkean committed:
migrating analytics module preference datastore to new proto datastore

rhkean committed:
migrate analytics preferences datastore to proto datastore

rhkean committed:
Merge branch 'main' into migrate_to_proto-datastore

rhkean committed:
reverting firebaseBom
